Orange/clustering/hierarchical.py

```diff
@@ -355,13 +355,14 @@ def optimal_leaf_ordering(tree, distances, progress_callback=None):
 
     :param Tree tree:
         Binary hierarchical clustering tree.
-    :param Orange.misc.DistMatrix distances:
-        DistMatrix that was used to compute the clustering.
+    :param numpy.ndarray distances:
+        A (N, N) numpy.ndarray of distances that were used to compute
+        the clustering.
     :param function progress_callback:
         Function used to report on progress.
 
     """
-    distances = distances.X
+    distances = numpy.asarray(distances)
```

Comment:
Before the change it was a DistMatrix. Now, let's say we are more general, but even if an exotic subclass is provided, it only needs to support indexing (line 370), then it is again converted to an (contiguous) array.

But the added check does not hurt since it does not do much in normal circumstances.